tests/shared.py

```diff
@@ -177,22 +179,25 @@ def pagination_tester(
 
 def make_request(
     stapi_client: TestClient,
-    endpoint: str,
+    url: str,
     method: str,
     body: dict | None,
-    next_token: str | None,
     limit: int,
 ) -> Response:
     """request wrapper for pagination tests"""
 
     match method:
         case "GET":
-            if next_token:  # extract pagination token
-                next_token = next_token.split("next=")[1]
-            params = {"next": next_token, "limit": limit}
-            res = stapi_client.get(endpoint, params=params)
+            o = urlparse(url)
```

Comment:
re-wrote this to actually parse the url, since a simple split on `next=` doesn't work if there are any other parameters, e.g., `next=foo&limit=1`, since the next_token value gets set to `foo&limit=1`